jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/FileBufferedResponseHandler.java

```diff
@@ -205,9 +206,9 @@ protected Action process() throws Exception
                     if (_last)
                         return Action.SUCCEEDED;
 
-                    long len = Math.min(MAX_MAPPED_BUFFER_SIZE, fileLength - _pos);
+                    long len = Math.min(MAX_BUFFER_SIZE, fileLength - _pos);
                     _last = (_pos + len == fileLength);
-                    ByteBuffer buffer = BufferUtil.toMappedBuffer(_filePath, _pos, len);
+                    ByteBuffer buffer = readByteBuffer(_filePath, _pos, len);
```

Comment:
`readByteBuffer()` could allocate a 1 GB buffer, each time a file is served! Yikes!

Reply:
If you have 1 GB of generated response body, then yup, and that's what the javadoc at the top of this handler warns about.

Reply:
Addressed this with the new internal `SendFileCallback` class, along with bufferSize configurable (defaults to 64kb)

Reply:
No longer relevant.